### ISSUE TITLE
get rid of external _decorators, mostly all the _numpy_testing ones.

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#9485]
I think we can mostly assume that when IPython is installed, user have pip and can get external packages.

Investigate what are the things we **actually** use and might be in numpy and see if we **really** need them 

Thoughts?

**Comment by takluyver** [comment]
I don't think we should make a hard dependency on numpy, but yes, I think we can probably get rid of some/all of that code.

**Comment by Carreau** [comment]
Already partially done, and more to come in #10326 

That's about all we can do without rewriting part of our test. Bumping to 7.0
